iso15118/secc/controller/evse_data.py

```diff
@@ -0,0 +1,126 @@
+from dataclasses import dataclass
+from typing import Optional, Union
+
+from iso15118.secc.controller.common import Limits
+
+
+@dataclass
+class EVSEACCLLimits(Limits):
+    # Optional in both Scheduled and Dynamic CL (both AC CL and BPT AC CL)
+    evse_target_active_power: Optional[float] = None  # Required in Dynamic AC CL
+    evse_target_active_power_l2: Optional[float] = None
+    evse_target_active_power_l3: Optional[float] = None
+    evse_target_reactive_power: Optional[float] = None
+    evse_target_reactive_power_l2: Optional[float] = None
+    evse_target_reactive_power_l3: Optional[float] = None
+    evse_present_active_power: Optional[float] = None  # Optional in AC CPD
+    evse_present_active_power_l2: Optional[float] = None  # Optional in AC CPD
+    evse_present_active_power_l3: Optional[float] = None  # Optional in AC CPD
+
+
+@dataclass
+class EVSEACLimits(Limits):
+    # 15118-2 AC CPD
+    evse_nominal_voltage: Optional[float] = None  # Also required for 15118-20 CPD
+    evse_max_current: Optional[float] = None  # Required
+
+    # 15118-20 AC CPD (Required)
+    evse_max_charge_power: Optional[float] = None
+    evse_min_charge_power: Optional[float] = None
+
+    # 15118-20 AC CPD (Optional)
+    evse_max_charge_power_l2: Optional[float] = None
+    evse_max_charge_power_l3: Optional[float] = None
+    evse_min_charge_power_l2: Optional[float] = None
+    evse_min_charge_power_l3: Optional[float] = None
+    evse_nominal_frequency: Optional[float] = None
+    max_power_asymmetry: Optional[float] = None
+    evse_power_ramp_limit: Optional[float] = None
+
+    evse_present_active_power: Optional[float] = None  # Optional in AC Scheduled CL
+    evse_present_active_power_l2: Optional[float] = None  # Optional in AC Scheduled CL
+    evse_present_active_power_l3: Optional[float] = None  # Optional in AC Scheduled CL
+
+
+@dataclass
+class EVSEACBPTLimits(Limits):
+    evse_max_discharge_power: Optional[float] = None  # Required
+    evse_max_discharge_power_l2: Optional[float] = None  # Optional
+    evse_max_discharge_power_l3: Optional[float] = None  # Optional
+
+    evse_min_discharge_power: Optional[float] = None  # Required
+    evse_min_discharge_power_l2: Optional[float] = None  # Optional
+    evse_min_discharge_power_l3: Optional[float] = None  # Optional
```

Comment:
do we actually need to segregate these two classes? 
why dont we just have the class `EVSEACCPDLimits` and add a comment `# BPT limits` and below the `evse_..._discharge..` fields?

But if we need the two classes, then shall we not subclass the `EVSEACBPTCPDLimits` from the `EVSEACCPDLimits` class?

Reply:
These will be direct mapping from cs_limits in pro land. So for now, I have kept the classes separate but have changed the BPT versions to inherit from the respective class.